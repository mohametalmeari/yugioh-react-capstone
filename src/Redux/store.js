import { configureStore } from '@reduxjs/toolkit';
import typesReducer from './types/typesSlice';
import detailsReducer from './details/detailsSlice';

const store = configureStore({
  reducer: {
    types: typesReducer,
    details: detailsReducer,
  },
});

export default store;
