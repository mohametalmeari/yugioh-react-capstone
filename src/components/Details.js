import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getCardsDetails } from '../Redux/details/detailsSlice';
import CardDetails from './CardDetails';
import headerImg from '../images/5.png';

const Details = () => {
  const type = window.location.pathname.split('/').slice(1).join('/').replace(/-/g, ' ');
  const { detailsList, isLoading } = useSelector((state) => state.details);
  const dispatch = useDispatch();
  useEffect(() => {
    console.log(type);
    dispatch(getCardsDetails(type));
    console.log(detailsList);
    console.log(isLoading);
  }, [type]);

  if (isLoading) {
    return (
      <div>
        <p>Loading...</p>
      </div>
    );
  }
  return (
    <div className="container">
      <header>
        <img className="header-img" src={headerImg} alt="Logo" />
        <div className="header-content">
          <h1>
            {type}
          </h1>
          <p>
            {detailsList.length}
            {' '}
            cards
          </p>
        </div>
      </header>
      <body>
        <h2 className="list-title">CARDS DETAILS</h2>
        <section className="details-list">
          {detailsList.map((item) => (
            <CardDetails
              key={item.id}
              cardsRace={item.race}
              cardsCount={item.count}
            />
          ))}
        </section>
      </body>
    </div>
  );
};
export default Details;
