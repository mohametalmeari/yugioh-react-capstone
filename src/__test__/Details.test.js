import { render, screen, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Details from '../components/Details';
import store from '../Redux/store';

describe('Cards Details Component', () => {
  beforeEach(() => {
    cleanup();
    delete window.location;
    window.location = { pathname: '/type-Union-Effect-Monster' };
  });
  afterEach(() => {
    cleanup();
  });
  test('renders race', async () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Details />
        </Provider>
      </MemoryRouter>,
    );
    expect(screen.getByText(/Type Details/i)).toBeInTheDocument();
    expect(screen.getByText(/Union Effect Monster/i)).toBeInTheDocument();
    expect(screen.getByText(/STATS BY RACE/i)).toBeInTheDocument();
  });
  test('renders header image', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Details />
        </Provider>
      </MemoryRouter>,
    );
    expect(screen.getByTestId('img-3')).toBeInTheDocument();
  });
  test('renders Loading', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Details />
        </Provider>
      </MemoryRouter>,
    );
    expect(screen.getByText(/Loading.../i)).toBeInTheDocument();
  });
  test('renders Details Page Snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Details />
        </Provider>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
