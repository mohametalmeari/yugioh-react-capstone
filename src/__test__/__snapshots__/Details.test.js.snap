// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Details Component renders Details Page Snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <nav>
      <a
        aria-current="page"
        class="nav-link active"
        href="/"
      >
        <svg
          class="w-6 h-6"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 19.5L8.25 12l7.5-7.5"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </a>
      <span>
        Type Details
      </span>
    </nav>
    <header>
      <img
        alt="Type"
        class="header-img"
        data-testid="img-3"
        src="1.png"
      />
      <div
        class="streched-gap"
      />
      <div
        class="header-content"
      >
        <h2
          class="races-title"
        >
          Union Effect Monster
        </h2>
        <p>
          0
           
          cards
        </p>
      </div>
    </header>
    <main>
      <h2
        class="list-title"
      >
        STATS BY RACE
      </h2>
      <section
        class="details-list"
      >
        <p>
          Loading...
        </p>
      </section>
    </main>
  </div>
</div>
`;
